### ISSUE TITLE
[BUG] Discarded page must be writable

### ISSUE THREAD

**Issue by jackdneilson** [litedb-org/LiteDB#2288]
**Version**
Which LiteDB version/OS/.NET framework version are you using. **(REQUIRED)**
v5.0.15 / Windows 10 / .NET Core 6

**Describe the bug**
Intermittently, when attempting to write to the database, an exception will be thrown in the form of
LiteDB ENSURE: Discarded page must be writable

**Code to Reproduce**
Write a small snippet to isolate your bug and could be possible to our team test. **(REQUIRED)**
// Note this code is called ~ 7 times each second from a single thread.
using (var db = new LiteDatabase(CONNECTION_STRING))
                            {
                                var col = db.GetCollection<Reading>();
                                col.Insert(new Reading()
                                {
                                    reading fields here
                                });
                            }

**Expected behavior**
Expected behaviour is for record to be inserted without exception.

**Screenshots/Stacktrace**
If applicable, add screenshots/stacktrace
The exception occurs on a small form factor PC with no network connection (writing to a local database file), so unfortunately I can't provide a stack trace.

**Additional context**
I'm not sure what causes the exception to be thrown as my application code is very simple, and as stated it happens intermittently. Typically the application will run for 4+ hours without issue, then crash due to an unhandled exception. Even when attempting to catch the exception this behaviour still occurs.

**Comment by jackdneilson** [comment]
@mbdavid apologies for the tag, do you have any suggestions for workarounds? This code is in production at the moment, and even wrapping it with a try-catch (Exception) block doesn't prevent it.

**Comment by BenWoodford** [comment]
I’ve just had this one pop up too, no idea what causes it and no idea how to resolve it… thankfully it was in a dev environment rather than with a customer but even so, I need to make sure it doesn’t happen reallt

**Comment by woi-123** [comment]
I'm also having the same issue, also intermittently occurring, with the exception message as follows:

LiteDB Version:5.0.8
CoreCLR Version: 4.700.21.56803
.NET Core Version: 3.1.22
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Exception: LiteDB ENSURE: discarded page must be writable
   at LiteDB.Constants.ENSURE(Boolean conditional, String message)
   at LiteDB.Engine.MemoryCache.DiscardPage(PageBuffer page)
   at LiteDB.Engine.DiskService.DiscardDirtyPages(IEnumerable`1 pages)
   at LiteDB.Engine.TransactionService.Dispose(Boolean dispose)
   at LiteDB.Engine.TransactionService.Finalize()

**Comment by rodrigojuarez** [comment]
I'm having this issue or something similiar with a db file hosted in Azure files, accessed by a containerized webapp.
page must be writable to support changes
LiteDB 5.0.20
Any suggestions?

**Comment by P-de-Jong** [comment]
@rodrigojuarez you had any luck with this yet?

**Comment by rodrigojuarez** [comment]
> @rodrigojuarez you had any luck with this yet?

Nah, I was never able to make it work with Azure files. Luckily for me, it was just a proof of concept to check if it would work. If you found any fix, please let me know!

**Comment by JKamsker** [comment]
This is fixed on the `dev` branch. `dev` ships as the LiteDB 6.x prereleases.

Please try the latest prerelease (currently **6.0.0-prerelease.81**):

```
dotnet add package LiteDB --prerelease
```

If the problem still happens on the prerelease, comment here with the version you used and a repro, and we'll reopen the issue.
